src/stream.cc

```diff
@@ -157,6 +157,18 @@ Result OutputBuffer::WriteToFile(std::string_view filename) const {
   return Result::Ok;
 }
 
+Result OutputBuffer::WriteToStdout() const {
```

Comment:
Why not call this `WriteToFile(FILE* file)` then it can be used both with `stdout_fileno` or with the result of fopen, and we can avoid duplicating the code between WriteToFile and WriteToStdout which do mostly the same thing.

Reply:
@sbc100 I considered doing that, but wasn't sure what to do with the error message in the shared part (which contains the filename): emitting the error message at the call site and doing an if error check there too (more code), or passing a filename into the shared part (sounds bad)

Reply:
OK sounds reasonable.